subsys/bluetooth/mesh/prov_device.c

```diff
@@ -92,8 +97,8 @@ static void prov_invite(const uint8_t *data)
 	/* Supported algorithms - FIPS P-256 Eliptic Curve */
 	net_buf_simple_add_be16(&buf, BIT(PROV_ALG_P256));
 
-	/* Public Key Type, Only "No OOB" Public Key is supported */
-	net_buf_simple_add_u8(&buf, PUB_KEY_NO_OOB);
+	/* Public Key Type */
+	net_buf_simple_add_u8(&buf, !!bt_mesh_prov->public_key_be);
```

Comment:
I know it'll be more lines of code, but I think it would be better to explicitly map to one of the allowed Public Key type values to make it clear that you're translating from some internal implementation detail (NULL pointer or not) to a valid provisioning protocol value.

So something like:
```
if (bt_mesh_prov->public_key_be == NULL) {
        net_buf_simple_add_u8(buf, PUB_KEY_NO_OOB);
} else {
        net_buf_simple_add_u8(buf, PUB_KEY_OOB);
}